src/pages/SignIn.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { memo } from 'react';
+
+import { Navigate } from 'react-router-dom';
+
+import { SubmitHandler, useForm } from 'react-hook-form';
+import useSession from 'hooks/useSession';
+import useDispatch from 'hooks/useDispatch';
+import useStatus from 'hooks/useStatus';
+
+import { signIn } from 'state/actions/userActions';
+
+import routes from 'components/routes/routes';
+import PageWrapper from 'components/common/PageWrapper';
+import InfoSection from 'components/common/InfoSection';
+import InputField from 'components/common/InputField';
+import BlackButton from 'components/common/BlackButton';
+
+import Smilies from 'assets/images/smilies.svg';
+
+import { PENDING } from 'constants/actionStatusConstants';
+
+import { signInFields } from 'types/sessionTypes';
+
+const SignIn = () => {
+  const { authenticated } = useSession();
+  const onSubmit = useDispatch(signIn);
+  const requestStatus = useStatus(signIn);
+
+  const { status, error } = requestStatus;
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<signInFields>();
+
+  const onSubmitForm: SubmitHandler<signInFields> = (data: signInFields) => {
+    onSubmit({ user: data });
+  };
+
+  if (authenticated) {
+    return <Navigate to={routes.index.path} />;
+  }
+
+  const isLoading = status === PENDING;
+
+  const extractEmailErrors = () => {
+    if (error) {
+      return error;
+    }
+    if (errors.email) {
+      return errors.email.message;
+    }
+    return '';
+  };
+
+  return (
+    <PageWrapper>
+      <>
```

Comment:
I think this fragment is unnecessary, right?

Reply:
Nope we need it 
![image](https://user-images.githubusercontent.com/41349687/163278370-a1a163b6-105f-482d-849c-fd4bd8092347.png)